### DISCUSSION TITLE
Automerge only when pipelines succeed

### DISCUSSION THREAD

**Discussion by Hussain-Ruhullah** [renovatebot/renovate#14357] [Request Help] [answered]
Is it possible to configure renovate to only merge updates when all pipeline i.e test, e2e are passed?

**Comment by rarkins** [comment]
That's the default behavior for Renovate's automerge - no configuration necessary. A repo with missing, pending, or failing tests will not automerge.

**Reply by rarkins** [reply, marked as answer]
So far your questions are all answered in https://docs.renovatebot.com/key-concepts/automerge/

If you are seeing something which is different to our documented behavior, please tell.

**Reply by Hussain-Ruhullah** [reply]
okay thank you Rarkins

**Reply by zbynek** [reply]
@rarkins do you have any idea why https://github.com/jenkins-infra/plugin-site/pull/1674 got merged? I don't think it ever was in a state where all pipelines were successful.

**Reply by viceice** [reply]
@zbynek please open a new discussion next time.

the auto merge was done by GitHub, because you allow it and Renovate uses it by default.
GitHub only waits for mandatory checks before merging, so your issue is related to branch protection.

check the docs for platform automerge

**Reply by zbynek** [reply]
Thanks!
